src/plugins/data/server/search/search_service.ts

```diff
@@ -140,6 +152,58 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
       ) => {
         return this.search(context, searchRequest, options);
       },
+      searchSource: {
+        asScoped: async (request: KibanaRequest) => {
+          const esClient = elasticsearch.client.asScoped(request);
+          const savedObjectsClient = savedObjects.getScopedClient(request);
+          const scopedIndexPatterns = await indexPatterns.indexPatternsServiceFactory(request);
+          const uiSettingsClient = uiSettings.asScopedToClient(savedObjectsClient);
+
+          // cache ui settings, only including items which are explicitly needed by SearchSource
+          const uiSettingsCache = pick(
+            await uiSettingsClient.getAll(),
+            searchSourceRequiredUiSettings
+          );
+
+          const searchSourceDependencies: SearchSourceDependencies = {
+            getConfig: <T = any>(key: string): T => uiSettingsCache[key],
+            search: (searchRequest, options) => {
+              /**
+               * Unless we want all SearchSource users to provide both a KibanaRequest
+               * (needed for index patterns) AND the RequestHandlerContext (needed for
+               * low-level search), we need to fake the context as it can be derived
+               * from the request object anyway. This will pose problems for folks who
+               * are registering custom search strategies as they are only getting a
+               * subset of the entire context. Ideally low-level search should be
+               * refactored to only require the needed dependencies: esClient & uiSettings.
+               */
+              const fakeRequestHandlerContext = {
+                core: {
+                  elasticsearch: {
+                    client: esClient,
+                  },
+                  uiSettings: {
+                    client: uiSettingsClient,
+                  },
+                },
+              } as RequestHandlerContext;
+              return this.search(fakeRequestHandlerContext, searchRequest, options);
+            },
+            // onResponse isn't used on the server, so we just return the original value
+            onResponse: (req, res) => res,
+            legacy: {
+              callMsearch: getCallMsearch({
+                esClient,
+                globalConfig$: this.initializerContext.config.legacy.globalConfig$,
+                uiSettings: uiSettingsClient,
+              }),
+              loadingCount$: new BehaviorSubject(0),
```

Comment:
`loadingCount$` isn't really necessary on the server, so we just initialize `BehaviorSubject` that never changes